x/zoneconcierge/abci.go

```diff
@@ -20,7 +20,13 @@ func BeginBlocker(ctx context.Context, k keeper.Keeper) error {
 func EndBlocker(ctx context.Context, k keeper.Keeper) ([]abci.ValidatorUpdate, error) {
 	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
 
-	k.BroadcastBTCHeaders(ctx)
-	k.BroadcastBTCStakingConsumerEvents(ctx)
+	// error in generating IBC packet data or sending packets is consensus-critical
+	if err := k.BroadcastBTCHeaders(ctx); err != nil {
+		panic(err)
+	}
+	if err := k.BroadcastBTCStakingConsumerEvents(ctx); err != nil {
+		panic(err)
+	}
```

Comment:
I think this is better than ignoring errors, but wouldn't panicking in the end blocker cause a chain stop? I mean, imagine if one of the BSNs is down for whatever reason, that will cause the Babylon chain to stop.

Not sure this is the best option. Perhaps logging the error here and continuing makes more sense, even if it'll lead to an inconsistent state between Babylon and the BSN.

Reply:
Thanks for clarifying!